### ISSUE TITLE
import flutter/flutter.h not found after upgrade flutter v2 on ios  #ios #not_found_import

### ISSUE THREAD

**Issue by oengmengthong** [flutter/flutter#79920]
### **1. remove folder ios:**

- Backup ios/Runner folder.
- Delete the ios folder.
- Run flutter create (your project name). in the previous folder where you have your project(cd users/user/"projects_folder") (this will recreate your ios folder).
- Paste your Runner backup in the ios folder (into the project).
- Open Runner.xcworkspace (into ios folder) and there, check the Version, the Bundle ID, all the info.
- (If do you Have Firebase, you have to copy and paste again the Google Service-Info.Plist into the Runner folder (Always through Xcode) (If do you do this manually, it doesn't work).

Finally, flutter run and should work!

If flutter run fails:

- `cd ios`
- `pod install`
- `cd ..`
- `flutter run`

### **2. clean up in ios folder:**

- `cd ios` for going to iOS directory of the flutter app
- `pod deintegrate`
- `rm Flutter/Flutter.podspec`
- `rm podfile.lock`
- `flutter clean`
- `flutter run`

### **3. Switch channel flutter to from stable to master:**

- `flutter channel master`
- `flutter upgrade --force`
- `flutter doctor -v`

### **4.remove Flutter.podspec only:**

- `rm ios/Flutter/Flutter.podspec`
- `flutter clean`

### **Here is error message:**

<details>
<summary>logs</summary>

```bash
> Failed to build iOS app
> Error output from Xcode build:
> ↳
>     ** BUILD FAILED **
> 
> Xcode's output:
> ↳
>     Command CompileSwift failed with a nonzero exit code
>     Command CompileSwift failed with a nonzero exit code
>     Command CompileSwift failed with a nonzero exit code
>     Command CompileSwift failed with a nonzero exit code
>     Command CompileSwift failed with a nonzero exit code
>     In file included from /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/JavaScriptChannelHandler.m:5:
>     /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/JavaScriptChannelHandler.h:5:9: fatal error: 'Flutter/Flutter.h'
>     file not found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     note: Using new build system
>     note: Building targets in parallel
>     note: Planning build
>     note: Constructing build description
> 
> Could not build the application for the simulator.
> Error launching application on iPhone 12 mini.

```

</details>

### **Here another message error:**

<details>
<summary>logs</summary>

```bash
> Failed to build iOS app
> Error output from Xcode build:
> ↳
>     ** BUILD FAILED **
> 
> 
> Xcode's output:
> ↳
>     In file included from /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/JavaScriptChannelHandler.m:5:
>     /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/JavaScriptChannelHandler.h:5:9: fatal error: 'Flutter/Flutter.h'
>     file not found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     1 error generated.
>     In file included from /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/FLTWebViewFlutterPlugin.m:5:
>     /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/FLTWebViewFlutterPlugin.h:5:9: fatal error: 'Flutter/Flutter.h'
>     file not found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     In file included from /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/FLTWKProgressionDelegate.m:5:
>     /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/FLTWKProgressionDelegate.h:5:9: fatal error: 'Flutter/Flutter.h'
>     file not found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     In file included from /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/FLTWKNavigationDelegate.m:5:
>     /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/FLTWKNavigationDelegate.h:5:9: fatal error: 'Flutter/Flutter.h'
>     file not found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     In file included from /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/FLTCookieManager.m:5:
>     /Users/admin/.pub-cache/hosted/pub.dartlang.org/webview_flutter-2.0.3/ios/Classes/FLTCookieManager.h:5:9: fatal error: 'Flutter/Flutter.h' file not
>     found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     Command CompileSwift failed with a nonzero exit code
>     Command CompileSwift failed with a nonzero exit code
>     Command CompileSwift failed with a nonzero exit code
>     /Users/admin/.pub-cache/hosted/pub.dartlang.org/video_player-2.1.0/ios/Classes/messages.m:8:9: fatal error: 'Flutter/Flutter.h' file not found
>     #import <Flutter/Flutter.h>
>             ^~~~~~~~~~~~~~~~~~~
>     1 error generated.
>     note: Using new build system
>     note: Building targets in parallel
>     note: Planning build
>     note: Constructing build description
> 
> Could not build the application for the simulator.
> Error launching application on iPhone 12 mini.

```

</details>

### **Here my flutter version**

<details>
<summary>flutter doctor -v</summary>

```bash
> [✓] Flutter (**Channel stable, 2.0.4, on macOS 11.2.3** 20D91 darwin-x64, locale en-KH)
>     • **Flutter version 2.0.4** at /Users/admin/flutter
>     • Framework revision b1395592de (5 days ago), 2021-04-01 14:25:01 -0700
>     • Engine revision 2dce47073a
>     • **Dart version 2.12.2**
> 
> [✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
>     • Android SDK at /Users/admin/Library/Android/sdk
>     • Platform android-30, build-tools 30.0.3
>     • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
>     • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
>     • All Android licenses accepted.
> 
> [✓] Xcode - develop for iOS and macOS
>     • Xcode at /Applications/Xcode.app/Contents/Developer
>     • Xcode 12.4, Build version 12D4e
>     • CocoaPods version 1.10.1
> 
> [✓] Chrome - develop for the web
>     • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome
> 
> [✓] Android Studio (version 4.1)
>     • Android Studio at /Applications/Android Studio.app/Contents
>     • Flutter plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/9212-flutter
>     • Dart plugin can be installed from:
>       🔨 https://plugins.jetbrains.com/plugin/6351-dart
>     • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
> 
> [✓] VS Code (version 1.55.0)
>     • VS Code at /Applications/Visual Studio Code.app/Contents
>     • Flutter extension version 3.21.0
> 
> [✓] Connected device (2 available)
>     • iPhone 12 mini (mobile) • 6DD0D762-592F-407B-87E3-168771D8441F • ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-4 (simulator)
>     • Chrome (web)            • chrome                               • web-javascript • Google Chrome 89.0.4389.90
> 
> • No issues found!
```

</details>

**Comment by oengmengthong** [comment]
I tried this solution also. but not working for me.

https://www.gitmemory.com/issue/flutter/flutter/24428/498702374

**Comment by imSanjaySoni** [comment]
### Facing same issue (  webview_flutter: ^1.0.7) 
```
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
lib/main.dart:1
Xcode build done.                                           12.4s
Failed to build iOS app
Error output from Xcode build:
↳
    objc[19770]: Class AMSupportURLConnectionDelegate is implemented in both ?? (0x2090d8188) and ?? (0x1252042b8). One of the two will be used. Which one is undefined.
    objc[19770]: Class AMSupportURLSession is implemented in both ?? (0x2090d81d8) and ?? (0x125204308). One of the two will be used. Which one is undefined.
    ** BUILD FAILED **
Xcode's output:
↳
    /Users/innobits/Desktop/new/cheetah/ios/Runner/GeneratedPluginRegistrant.m:130:9: fatal error: module 'webview_flutter' not found
    @import webview_flutter;
     ~~~~~~~^~~~~~~~~~~~~~~
    1 error generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

```

**Comment by oengmengthong** [comment]
i found 1 more solution but still got error 

[#70895 (comment)](https://github.com/flutter/flutter/issues/70895#issuecomment-744734693)

`$ rm ios/Flutter/Flutter.podspec
$ flutter clean`

**Comment by tsuruobiz** [comment]
@oengmengthong 

Same problem for me in stable channel `2.0.4`. 
I confused so long time too, but downgrade to `2.0.3`, and then works fine.
try it?
- `flutter downgrade`

**Comment by jmagman** [comment]
@oengmengthong  Can you attach your full `flutter run` logs with the `--verbose` flag?  It will be large, you'll need to attach the file.

**Comment by oengmengthong** [comment]
my issues fixed by: flutter v2.0.4 with this solution

1. remove folder ios:
Backup ios/Runner folder.
Delete the ios folder.
Run flutter create (your project name). in the previous folder where you have your project(cd users/user/"projects_folder") (this will recreate your ios folder).
Paste your Runner backup in the ios folder (into the project).
Open Runner.xcworkspace (into ios folder) and there, check the Version, the Bundle ID, all the info.
(If do you Have Firebase, you have to copy and paste again the Google Service-Info.Plist into the Runner folder (Always through Xcode) (If do you do this manually, it doesn't work).
Finally, flutter run and should work!

If flutter run fails:

```
cd ios
pod install
cd ..
flutter run
```

**Comment by DotSyndicate** [comment]
Hi these solutions aren't working for me. Is there any other solution. Why has this suddenly started appearing?

**Comment by jmagman** [comment]
@DotSyndicate Can you file a new issue, completely fill out the template, and attach the full logs from a `flutter run` with the `--verbose` flag?

**Comment by DotSyndicate** [comment]
Yes. I will file a new issue but apparently, everyone will point to this and other issues filed here. However, I will file a new issue.
